### PR TITLE
Use nonNullish for not undefined comparison

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

As for `isNullish`, refactor not undefined comparison with `nonNullish` utils.

### CODE REVIEW

File: src/frontend/src/utils/userNumber.test.ts [hunk 1]
```diff
@@ -90,11 +91,11 @@ function testLocalStorage(
 ) {
   test(name, () => {
     localStorage.clear();
-    if (opts?.before !== undefined) {
+    if (nonNullish(opts) && nonNullish(opts?.before)) {
```
Comment (nmattia) [inline src/frontend/src/utils/userNumber.test.ts:94]: why is the first check necessary?

Comment (peterpeterparker) [reply src/frontend/src/utils/userNumber.test.ts:94]: Capture d’écran 2023-04-29 à 11 30 29

File: src/frontend/src/utils/userNumber.test.ts [hunk 2]
```diff
@@ -90,11 +91,11 @@ function testLocalStorage(
...
       setLocalStorage(opts.before);
```
Comment (nmattia) [inline src/frontend/src/utils/userNumber.test.ts:95]:
```suggestion
    const before = opts?.before;
    if (nonNullish(before)) {
      setLocalStorage(before);
```


what do you think about this? I think it makes more sense

**Review by nmattia — APPROVED** [review]

File: src/frontend/src/utils/userNumber.test.ts [hunk 3]
```diff
@@ -90,11 +91,12 @@ function testLocalStorage(
...
     localStorage.clear();
-    if (opts?.before !== undefined) {
-      setLocalStorage(opts.before);
+    const before = opts?.before;
+    if (nonNullish(before)) {
+      setLocalStorage(before);
     }
     fn();
-    if (opts?.after !== undefined) {
+    if (nonNullish(opts) && nonNullish(opts?.after)) {
```
Comment (nmattia) [inline src/frontend/src/utils/userNumber.test.ts:99]: would it make sense to do the same for `after` as we do with `before`?

Comment (peterpeterparker) [reply src/frontend/src/utils/userNumber.test.ts:99]: sure go for it

**Pull request merged**